### ISSUE TITLE
New create index check breaks migration generation for Atlas

### ISSUE THREAD

**Issue by mathieu-lemay** [go-gorm/postgres#291]
## GORM Playground Link
https://github.com/go-gorm/playground/pull/774

## Description
Since PR #285, the migration generation for Atlas fails. This is likely due to the generation using a mock database that doesn't really have the index.
Since that change, I get some `failed to create index ...` errors

### How to reproduce
This sample program shows the issue. With `go-gorm/postgres@v1.5.9`, it works fine, but it fails with `go-gorm/postgres@v1.5.10`.

```go
package main

import (
        "io"
        "log"
        "os"

        "ariga.io/atlas-provider-gorm/gormschema"

        "gorm.io/gorm"
)

type Product struct {
  gorm.Model

  Code  string
  Price uint
}

func main() {
        stmts, err := gormschema.New("postgres").Load(&Product{})

        if err != nil {
                log.Fatalf("error loading gorm schema: %v\n", err)
        }

        _, err = io.WriteString(os.Stdout, stmts)

        if err != nil {
                log.Fatalf("error writing statements to stdout: %v\n", err)
        }
}
```

Expected behavior: It should output a SQL script to create the Products table
Actual behavior: It fails with `error loading gorm schema: failed to create index with name idx_products_deleted_at`

**Comment by BarAdar4** [comment]
Same issue happens for me as well. Works with `go-gorm/postgres@v1.5.9`, fails in `go-gorm/postgres@v1.5.10`, with the error message `failed to create index ... `

**Comment by samirsss** [comment]
We hit the issue too and hopefully this will get fixed soon - thanks

**Comment by giautm** [comment]
Hello all, Atlas' Integration team here. Honestly, I don't think the fix in https://github.com/go-gorm/postgres/pull/285 is right. As its description said:
> Check index exists after create index. If not exists, return the err.

The migrator engine is suppose to run the migration script and the DB have to execute it and report any error. In this default Migrator, it is using: `IF NOT EXISTS ? ON ?` will cause the DB ignore the second DDL for the same index name if it already exists. So, what's the benefit of adding `HasIndex()` after execute the DDL? Do we know the existing index is updated or not?

> We run into this case, because we use the same index name for difference table, and the migration executed successfuly. 

This is the application error, because it use the same name for both index, please fix it in your application code. And the PR https://github.com/go-gorm/postgres/pull/285 should be revert.

Thank you,

cc @jinzhu

**Comment by a8m** [comment]
> We run into this case, because we use the same index name for difference table, and the migration executed successfuly.

How is this possible? The index name must be unique in the schema, so no other index, table, view, etc., can use the same name. Also, if the DDL was executed successfully, the engine should rely on the database to create it, and we should not verify or test Postgres here.

This sounds to me like there’s an incorrect usage of search_path by the PR author, and the PR should be reverted.

**Comment by a8m** [comment]
@mathieu-lemay, the issue was resolved with #292.
